ibtk/include/ibtk/private/ScopedVectorDuplicate-inl.h

```diff
@@ -0,0 +1,97 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2023 - 2023 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBTK_ScopedVectorDuplicate_inl_h
+#define included_IBTK_ScopedVectorDuplicate_inl_h
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibtk/config.h>
+
+#include <ibtk/ScopedVectorDuplicate.h>
+#include <ibtk/ibtk_utilities.h>
+
+#include <tbox/Pointer.h>
+
+#include <SAMRAIVectorReal.h>
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBTK
+{
+template <typename TYPE>
+ScopedVectorDuplicate<TYPE>::ScopedVectorDuplicate(
+    const SAMRAI::tbox::Pointer<SAMRAI::solv::SAMRAIVectorReal<NDIM, TYPE> >& vector,
+    const std::string& name)
+    : ScopedVectorDuplicate(checked_dereference(vector), name)
+{
+}
+
+template <typename TYPE>
+ScopedVectorDuplicate<TYPE>::ScopedVectorDuplicate(const SAMRAI::solv::SAMRAIVectorReal<NDIM, TYPE>& vector,
+                                                   const std::string& name)
+{
+    d_vector = vector.cloneVector(name);
+    d_vector->allocateVectorData();
```

Comment:
Do we want to allow for users to pass in a value for the `timestamp` argument to `allocateVectorData()`?

Reply:
We could, but I don't think its necessary - these vectors should not participate in time interpolation.